jobs.go

```diff
@@ -12,18 +12,22 @@ import (
 
 // Job holds a single job to be run
 type Job struct {
-	Name     string            `yaml:"name"`
-	URL      string            `yaml:"url"`
-	Method   string            `yaml:"method"`
-	Type     string            `yaml:"type"`
-	Interval uint16            `yaml:"interval"`
-	Cookies  []Cookie          `yaml:"cookies,omitempty"`
-	Tags     map[string]string `yaml:"tags,omitempty"`
+	Step     JobStep   `yaml:",inline"`
+	Type     string    `yaml:"type"`
+	Interval uint16    `yaml:"interval"`
+	Steps    []JobStep `yaml:"steps,omitempty"` // an api job may consist of multiple jobs (steps)
 }
 
-// JobConfig holds a list of jobs to be run
-type JobConfig struct {
-	Jobs []Job `yaml:"jobs"`
+type JobStep struct {
+	Name    string              `yaml:"name"`
+	URL     string              `yaml:"url"`
+	Method  string              `yaml:"method"`
+	Cookies []Cookie            `yaml:"cookies,omitempty"`
+	Header  map[string][]string `yaml:"header,omitempty"`
+	// if header contains a different content type this overwrites it.
+	ContentType string            `yaml:"content-type,omitempty"`
+	Body        string            `yaml:"body,omitempty"`
+	Tags        map[string]string `yaml:"tags,omitempty"`
```

Comment:
This should probably be ripped out into its own class file and we can add a step runner class so we don't have duplicate code in api and simple. We can do that in a separate PR though.